tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -852,7 +859,31 @@ const reducer = createReducer(
         }
       }
 
-      const nextState = {
+      const nextCardMetadataList = buildCardMetadataList(state.tagMetadata);
```

Comment:
My  understanding (which, admittedly, is weak and based on looking at metrics effects and data source code this morning) is that  a fetchTimeSeriesLoaded will only ever correspond to exactly one card.

It looks like here you are rebuilding the min/max for ALL cards on each response. Is that necessary? Can we limit it to only rebuilding min/max for the card represented in the response?

Reply:
I believe I've refactored this correctly AND made it more efficient at the same time.